BUG: test_convergence fails very often on aarch64-linux

### Describe your issue.

Trying to build and test latest SciPy on NixOS unstable, aarch64-linux, results in the test failing, seemingly with hardware-dependent results.

Failure logs (RK3588 SBC):

```
_________ test_convergence[rand-sym-pd-F-cgs-numpy-batch_b0-batch_A0] __________
[gw6] linux -- Python 3.14.6 /nix/store/sfbjq7mllid60l3jvxw0q2gknqldyy2l-python3-3.14.6/bin/python3.14

case = namespace(name='rand-sym-pd-F-cgs', A=array([[3.5522242, 2.3749914, 2.6603491, 3.117838 , 2.7458262, 1.6847214,
      ..., 5., 6., 7., 8.], dtype=float32), convergence=True, solver=<function cgs at 0xffffe55f2fb0>, casename='rand-sym-pd-F')
xp = <module 'scipy._external.array_api_compat.numpy' from '/nix/store/0ljrm7xnxqz3qxpvllj4s8k0srb6rzdm-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/_external/array_api_compat/numpy/__init__.py'>
batch_A = (), batch_b = ()

    @pytest.mark.parametrize("batch_A", [()])
    @pytest.mark.parametrize("batch_b", [()])
    def test_convergence(case, xp, batch_A, batch_b):
        if (case.solver is tfqmr) and ("poisson2d-F" in case.name):
            pytest.skip("Struggles to converge with single precision on some platforms")
        case = xp_case(case, xp, batch_A, batch_b, rng=38)
        A = case.A

        if A.dtype in (xp.float64, xp.complex128):
            rtol = 1e-8
        else:
            rtol = 1e-2

        b = case.b
        x0 = 0 * b

        x, info = case.solver(A, b, x0=x0, rtol=rtol)
        xp_assert_equal(x0, 0 * b)  # ensure that x0 is not overwritten

        if case.convergence:
>           assert info == 0
E           assert 90 == 0

lib/python3.14/site-packages/scipy/sparse/linalg/_isolve/tests/test_iterative.py:337: AssertionError
```

Failure logs (Ampere Altra build server):

```
______ test_convergence[rand-cmplx-sym-pd-F-cgs-numpy-batch_b0-batch_A0] _______
[gw9] linux -- Python 3.14.6 /nix/store/sfbjq7mllid60l3jvxw0q2gknqldyy2l-python3-3.14.6/bin/python3.14

case = namespace(name='rand-cmplx-sym-pd-F-cgs', A=array([[7.5233746+4.5698533e-17j, 5.4297857-1.0201480e-01j,
        5.9857...  8.+0.j], dtype=complex64), convergence=True, solver=<function cgs at 0xffffe5652fb0>, casename='rand-cmplx-sym-pd-F')
xp = <module 'scipy._external.array_api_compat.numpy' from '/nix/store/0ljrm7xnxqz3qxpvllj4s8k0srb6rzdm-python3.14-scipy-1.18.0/lib/python3.14/site-packages/scipy/_external/array_api_compat/numpy/__init__.py'>
batch_A = (), batch_b = ()

    @pytest.mark.parametrize("batch_A", [()])
    @pytest.mark.parametrize("batch_b", [()])
    def test_convergence(case, xp, batch_A, batch_b):
        if (case.solver is tfqmr) and ("poisson2d-F" in case.name):
            pytest.skip("Struggles to converge with single precision on some platforms")
        case = xp_case(case, xp, batch_A, batch_b, rng=38)
        A = case.A
    
        if A.dtype in (xp.float64, xp.complex128):
            rtol = 1e-8
        else:
            rtol = 1e-2
    
        b = case.b
        x0 = 0 * b
    
        x, info = case.solver(A, b, x0=x0, rtol=rtol)
        xp_assert_equal(x0, 0 * b)  # ensure that x0 is not overwritten
    
        if case.convergence:
>           assert info == 0
E           assert 90 == 0

lib/python3.14/site-packages/scipy/sparse/linalg/_isolve/tests/test_iterative.py:337: AssertionError
```

### Reproducing Code Example

```python
https://github.com/scipy/scipy/blob/main/scipy/sparse/linalg/_isolve/tests/test_iterative.py#L319
```

### Error message

```shell
Included above
```

### SciPy/NumPy/Python version and system information

```shell
SciPy 1.18.0, NumPy 2.5.0, GCC 15.2.0
```

--- a/scipy/sparse/linalg/_isolve/tests/test_iterative.py
+++ b/scipy/sparse/linalg/_isolve/tests/test_iterative.py
@@ -37,10 +37,6 @@ _SOLVERS = [bicg, bicgstab, cg, cgs, gcrotmk, gmres, lgmres,
 CB_TYPE_FILTER = ".*called without specifying `callback_type`.*"
 
 
-def _is_32bit():
-    return np.intp(0).itemsize < 8
-
-
 def _assert_success(*, A, x, b, xp, rtol=1.0, atol=0.0, less_equal=False):
     Ax = xp.squeeze(A @ x[..., xp.newaxis], axis=-1)
     residual = Ax - b
@@ -323,8 +319,11 @@ def test_maxiter(case, xp, batch_A, batch_b):
 def test_convergence(case, xp, batch_A, batch_b):
     if (case.solver is tfqmr) and ("poisson2d-F" in case.name):
         pytest.skip("Struggles to converge with single precision on some platforms")
-    if (case.solver is tfqmr) and ("rand-sym-pd-F" in case.name) and _is_32bit():
-        pytest.skip("Fails to converge on i686 (32-bit) linux")
+    if (case.solver is tfqmr) and ("rand-sym-pd-F" in case.name):
+        # Numerical stability is borderline for single precision, tfqmr stagnates at a
+        # relative residual ||b - Ax||/||b|| of ~2e-2, failure yes/no depends on
+        # platform-specific rounding behavior (see, e.g., scipy#25522)
+        pytest.skip("Fails to converge with single precision on some platforms")
 
     case = xp_case(case, xp, batch_A, batch_b, rng=38)
     A = case.A
@@ -357,8 +356,9 @@ def test_precond_dummy(case, xp, batch_A, batch_b):
         pytest.skip("Struggles to converge with single precision")
     if (case.solver is tfqmr) and ("poisson2d-F" in case.name):
         pytest.skip("Hits divide-by-zero with single precision")
-    if (case.solver is tfqmr) and ("rand-sym-pd-F" in case.name) and _is_32bit():
-        pytest.skip("Fails to converge on i686 (32-bit) linux")
+    if (case.solver is tfqmr) and ("rand-sym-pd-F" in case.name):
+        # Numerical stability is borderline for float32 (see, scipy#25522)
+        pytest.skip("Fails to converge with single precision on some platforms")
     if not case.convergence:
         pytest.skip("Solver - Breakdown case, see gh-8829")
 

